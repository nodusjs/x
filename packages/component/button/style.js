import { css } from "@nodusjs/std/dom";

export const style = (button) =>
  css`
    :host {
      width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

      button {
        align-items: center;
        aspect-ratio: ${button.onlyIcon ? "1/1" : "auto"};
        background-color: var(--button-background-color-${button.color}-${button.variant});
        border: var(--button-border-${button.color}-${button.variant});
        border-radius: var(--radius-md);
        box-sizing: border-box;
        color: var(--button-color-${button.color}-${button.variant});
        cursor: pointer;
        display: inline-flex;
        font-family: var(--font-family-base);
        font-size: var(--button-font-size-${button.size});
        font-weight: var(--font-weight-semibold);
        gap: var(--button-gap-${button.size});
        height: var(--button-height-${button.size});
        justify-content: center;
        line-height: 1;
        padding: 0 ${button.onlyIcon ? 0 : "var(--spacing-lg)"};
        transition: all 0.2s ease-out;
        width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

        &:hover {
          background-color: var(--button-background-color-${button.color}-${button.variant}_hover);
          color: var(--button-color-${button.color}-${button.variant}_hover);
        }

        &:focus-visible {
          box-shadow: var(--button-box-shadow-${button.color});
          outline: 0;
        }
      }
    }

    :host(:state(disabled)) {
      button {
        background-color: var(--bg-disabled);
        border: 1px solid var(--border-disabled_subtle);
        color: var(--fg-disabled);
        cursor: auto;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
