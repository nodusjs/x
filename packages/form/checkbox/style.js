import { css } from "@nodusjs/std/dom";

export const style = () =>
  css`
    :host {
      align-items: start;
      box-sizing: border-box;
      display: flex;
      gap: var(--spacing-sm);
      width: var(--width-xxs);

      wrap {
        align-items: start;
        display: flex;
        flex-direction: column;
        gap: var(--spacing-sm);
      }

      input {
        background-color: var(--bg-primary);
        border: 1px solid var(--border-primary);
        border-radius: var(--radius-md);
        box-sizing: border-box;
        display: block;
        height: 16px;
        margin: 4px 0 0 0;
        resize: none;
        width: 16px;

        &:checked {
          accent-color: var(--bg-brand-solid);
        }

        &:focus-visible {
          box-shadow: var(--focus-ring);
          outline: 0;
        }

        &:active,
        &:hover {
          outline: 0;
        }

        &:disabled,
        &:read-only {
          background-color: var(--bg-secondary);
          border-color: var(--border-primary);
          box-shadow: none;
        }

        &::-webkit-autofill,
        &::-webkit-autofill:hover, 
        &::-webkit-autofill:focus, 
        &::-webkit-autofill:active {
          transition: background-color 9999999999s ease-in-out 0s
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }

    :host(:state(invalid)) {
      input {
        border-color: var(--border-error);
      }
    }
  `;
