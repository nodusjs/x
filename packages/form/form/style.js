import { css } from "@nodusjs/std/dom";

export const style = (form) =>
  css`
    :host {
      box-sizing: border-box;
      display: block;
      width: ${form.width};
      width: var(--width-${form.width}, ${form.width});

      form {
        container-type: inline-size;
        width: 100%;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
