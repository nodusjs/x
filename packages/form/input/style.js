import { css } from "@nodusjs/std/dom";

export const style = (input) =>
  css`
    :host {
      align-items: start;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      gap: var(--spacing-sm);
      width: ${input.width};
      width: var(--width-${input.width}, ${input.width});

      input {
        appearance: none;
        background-color: var(--bg-primary);
        border: 1px solid var(--border-primary);
        border-radius: var(--radius-md);
        box-sizing: border-box;
        color: var(--text-primary);
        font-family: var(--font-family-base);
        font-size: var(--font-size-text-md);
        font-weight: var(--font-weight-regular);
        height: 40px;
        line-height: var(--line-height-text-md);
        padding: var(--spacing-md) var(--spacing-lg);
        width: 100%;


        &[type="number"] {
          -moz-appearance: textfield;
        }

        &:active,
        &:hover {
          outline: 0;
        }

        &:focus {
          border-color: var(--border-brand);
          outline: 0;
        }

        &:disabled,
        &:read-only {
          background-color: var(--bg-secondary);
          border-color: var(--border-primary);
          box-shadow: none;
          color: var(--text-disabled);
        }

        &::placeholder {
          color: var(--text-placeholder);
        }

        &::-webkit-autofill,
        &::-webkit-autofill:hover, 
        &::-webkit-autofill:focus, 
        &::-webkit-autofill:active {
          transition: background-color 9999999999s ease-in-out 0s
        }

        &::-webkit-outer-spin-button,
        &::-webkit-inner-spin-button {
          -webkit-appearance: none;
          margin: 0;
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }

    :host(:state(invalid)) {
      input {
        border-color: var(--border-error);
        padding-right: 40px;
      }
    }
  `;
