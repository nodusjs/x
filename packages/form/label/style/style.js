import { css } from "@nodusjs/std/dom";

export const style = () =>
  css`
    :host {
      color: var(--text-secondary);
      font-family: var(--font-family-base);
      font-size: var(--font-size-text-sm);
      font-weight: var(--font-weight-medium);
      line-height: var(--line-height-text-sm); 
    }
  `;
