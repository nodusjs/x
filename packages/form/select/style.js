import { css } from "@nodusjs/std/dom";

export const style = (select) =>
  css`
    :host {
      align-items: start;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      gap: var(--spacing-sm);
      width: ${select.width};
      width: var(--width-${select.width}, ${select.width});


      wrap {
        position: relative;
        width: inherit;

        select {
          appearance: none;
          background-color: var(--bg-primary);
          border: 1px solid var(--border-primary);
          border-radius: var(--radius-md);
          box-sizing: border-box;
          color: var(--text-primary);
          font-family: var(--font-family-base);
          font-size: var(--font-size-text-md);
          font-weight: var(--font-weight-regular);
          height: 40px;
          line-height: var(--line-height-text-md);
          padding: var(--spacing-md) var(--spacing-5xl) var(--spacing-md) var(--spacing-lg);
          width: 100%;


          &:active,
          &:hover {
            outline: 0;
          }

          &:focus {
            border-color: var(--border-brand);
            outline: 0;
          }

          &:disabled,
          &:[readonly] {
            background-color: var(--bg-secondary);
            border-color: var(--border-primary);
            box-shadow: none;
            color: var(--text-disabled);
          }

          &::placeholder {
            color: var(--text-placeholder);
          }

          &::-webkit-autofill,
          &::-webkit-autofill:hover, 
          &::-webkit-autofill:focus, 
          &::-webkit-autofill:active {
            transition: background-color 9999999999s ease-in-out 0s
          }

          &::-webkit-outer-spin-button,
          &::-webkit-inner-spin-button {
            -webkit-appearance: none;
            margin: 0;
          }
        }

        x-icon {
          color: var(--fg-quaternary);
          position: absolute;
          right: var(--spacing-md);
          top: var(--spacing-md);
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }

    :host(:state(invalid)) {
      select {
        border-color: var(--border-error) !important;
      }
    }
  `;
