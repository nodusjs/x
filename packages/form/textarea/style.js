import { css } from "@nodusjs/std/dom";

export const style = () =>
  css`
    :host {
      align-items: start;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      gap: var(--spacing-sm);
      width: var(--width-xxs);

      textarea {
        appearance: none;
        background-color: var(--bg-primary);
        border: 1px solid var(--border-primary);
        border-radius: var(--radius-md);
        box-sizing: border-box;
        color: var(--text-primary);
        font-family: var(--font-family-base);
        font-size: var(--font-size-text-md);
        font-weight: var(--font-weight-regular);
        height: auto;
        line-height: var(--line-height-text-md);
        min-height: 128px;
        overflow: hidden;
        padding: var(--spacing-md) var(--spacing-lg);
        resize: none;
        width: 100%;

        &:active,
        &:hover {
          outline: 0;
        }

        &:focus {
          border-color: var(--border-brand);
          outline: 0;
        }

        &:disabled,
        &:read-only {
          background-color: var(--bg-secondary);
          border-color: var(--border-primary);
          box-shadow: none;
          color: var(--text-disabled);
        }

        &::placeholder {
          color: var(--text-placeholder);
        }

        &::-webkit-autofill,
        &::-webkit-autofill:hover, 
        &::-webkit-autofill:focus, 
        &::-webkit-autofill:active {
          transition: background-color 9999999999s ease-in-out 0s
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }

    :host(:state(invalid)) {
      textarea {
        border-color: var(--border-error);
        padding-right: 40px;
      }
    }
  `;
