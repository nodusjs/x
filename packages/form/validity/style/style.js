import { css } from "@nodusjs/std/dom";

/**
 * Define os estilos de exibição para o `<x-validity>`, incluindo cores,
 * tipografia e visibilidade condicional.
 *
 * @returns {CSSStyleSheet} Folha de estilo contendo regras para o host e para
 * o estado `invalid`.
 *
 * @description
 * Esta função retorna um template de CSS que esconde o elemento por padrão
 * e exibe-o apenas quando o estado de invalidade (`:host(:state(invalid))`)
 * estiver ativo. Também aplica variáveis de design system para cor, fonte,
 * peso e espaçamento, garantindo consistência visual.
 *
 * @example
 * import { style } from "./style.js";
 *
 * // Em um contexto de Shadow DOM:
 * const sheet = style();
 * shadowRoot.adoptedStyleSheets = [sheet];
 */
export const style = () => css`
  :host {
    color: var(--text-error-primary);
    display: none;
    font-family: var(--font-family-base);
    font-size: var(--font-size-text-sm);
    font-weight: var(--font-weight-regular);
    line-height: var(--line-height-text-sm);
  }

  :host(:state(invalid)) {
    display: inline;
  }
`;
