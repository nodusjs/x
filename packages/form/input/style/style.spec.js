import { describe, expect, it } from "vitest";
import { style } from "./style.js";

describe("style", () => {
  it("gera o bloco de styles CSS esperado", () => {
    const {
      cssRules: [{ cssText }],
    } = style();
    expect(cssText).toMatchSnapshot();
  });
});
