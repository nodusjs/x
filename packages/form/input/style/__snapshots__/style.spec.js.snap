// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`style > gera o bloco de styles CSS esperado 1`] = `":host {  }"`;
